src/rust/node-identifier/tests/integration_test.rs

```diff
@@ -0,0 +1,239 @@
+#![cfg(feature = "new_integration_tests")]
+
+use std::time::Duration;
+
+use bytes::Bytes;
+use futures::StreamExt;
+use kafka::{
+    Consumer,
+    ConsumerError,
+};
+use opentelemetry::{
+    global,
+    sdk::propagation::TraceContextPropagator,
+};
+use rust_proto_new::{
+    graplinc::grapl::{
+        api::{
+            graph::v1beta1::{
+                IdentifiedGraph,
+                IdentifiedNode,
+                ImmutableUintProp,
+                Property,
+            },
+            pipeline_ingress::v1beta1::{
+                client::PipelineIngressClient,
+                PublishRawLogRequest,
+            },
+        },
+        pipeline::v1beta2::Envelope,
+    },
+    protocol::healthcheck::client::HealthcheckClient,
+};
+use test_context::{
+    test_context,
+    AsyncTestContext,
+};
+use tokio::sync::oneshot;
+use tracing::Instrument;
+use tracing_appender::non_blocking::WorkerGuard;
+use tracing_subscriber::{
+    prelude::*,
+    EnvFilter,
+};
+use uuid::Uuid;
+
+fn find_node<'a>(
+    graph: &'a IdentifiedGraph,
+    o_p_name: &str,
+    o_p_value: Property,
+) -> Option<&'a IdentifiedNode> {
+    graph.nodes.values().find(|n| {
+        n.properties.iter().any(|(p_name, p_value)| {
+            p_name.as_str() == o_p_name && p_value.property.clone() == o_p_value
+        })
+    })
+}
+
+struct NodeIdentifierTestContext {
+    pipeline_ingress_client: PipelineIngressClient,
+    bootstrap_servers: String,
+    sasl_username: String,
+    sasl_password: String,
+    consumer_group_name: String,
+    _guard: WorkerGuard,
+}
```

Comment:
Might make sense to make these env vars - and the ones in main() - a Clap struct (formerly structopt)?

Reply:
What's the benefit? AFAIK we _only_ set these values via environment variables passed in via the Nomad HCL configs (not, e.g. in a `.properties` file or whatever). Does a Clap struct make this better?